### PR TITLE
Warn if firewalld service is enabled.

### PR DESCRIPTION
**Pull request by dgoodwin**
Fixes https://github.com/kubernetes/kubeadm/issues/21

Output will be:

```
(root@centos1 ~) $ kubeadm init             
Running pre-flight checks
WARNING: firewalld is active, please ensure ports [6443 9898 10250] are open
```

I went with the port list from @errordeveloper 's ansible playbook here but it's possible there should be others listed.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37498)

### CODE REVIEW

**Comment by dgoodwin** [comment]
@kubernetes/sig-cluster-lifecycle

File: cmd/kubeadm/app/preflight/checks.go [hunk 1]
```diff
@@ -80,6 +80,32 @@ func (sc ServiceCheck) Check() (warnings, errors []error) {
...
+
+	warnings = []error{}
+
+	if !initSystem.ServiceExists("firewalld") {
+		return nil, nil
+	}
+
+	if initSystem.ServiceIsActive("firewalld") {
+		warnings = append(warnings,
+			fmt.Errorf("firewalld is active, please ensure ports %v are open",
```
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: `firewalld is active, please ensure ports %v are open, otherwise the setup will fail`

Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: (Just so it sounds a little bit more scary and serious :smile:)

File: cmd/kubeadm/app/preflight/checks.go [hunk 2]
```diff
@@ -220,6 +246,7 @@ func RunInitMasterChecks(cfg *kubeadmapi.MasterConfiguration) error {
 		HostnameCheck{},
 		ServiceCheck{Service: "kubelet"},
 		ServiceCheck{Service: "docker"},
+		FirewalldCheck{ports: []int{int(cfg.API.BindPort), int(cfg.Discovery.BindPort), 10250}},
```
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: How about the other ports listed here below as `PortOpenCheck`s?

**Review by luxas — COMMENTED** [review]
Overall LGTM but a nit and a question

File: cmd/kubeadm/app/preflight/checks.go (hunk 2 above)
Comment (dgoodwin) [inline cmd/kubeadm/app/preflight/checks.go]: I am quite unsure, I suspect we don't need etcd externally accessible, any idea if all the others are valid?

File: cmd/kubeadm/app/preflight/checks.go (hunk 1 above)
Comment (dgoodwin) [inline cmd/kubeadm/app/preflight/checks.go]: I think init will succeed even in this case, but join will fail, how about "firewalld is active, please ensure ports %v are open or your cluster may not function correctly"

File: cmd/kubeadm/app/preflight/checks.go (hunk 2 above)
Comment (luxas) [inline cmd/kubeadm/app/preflight/checks.go]: Ah, it's only for externally accessible ports? Right, then these are just fine

File: cmd/kubeadm/app/preflight/checks.go (hunk 1 above)
Comment (dgoodwin) [inline cmd/kubeadm/app/preflight/checks.go]: Message updated.

**Comment by luxas** [comment]
LGTM, thanks!

**Pull request merged**